ivadomed/prepare_dataset.py

```diff
@@ -0,0 +1,213 @@
+    :return:
+    Args:
+        path_label: path of nifti image
+        aim: 'full' or 'c2' full will return all points with label between 3 and 30 , c2 will return only the coordinates of points label 3
+
+    Returns:
+        array: array containing the asked point in the format [x,y,z,value]
+
+    """
+    image = nib.load(path_label)
+    image = nib.as_closest_canonical(image)
+    arr = np.array(image.dataobj)
+    list_label_image = []
+    for i in range(len(arr.nonzero()[0])):
+        x = arr.nonzero()[0][i]
+        y = arr.nonzero()[1][i]
+        z = arr.nonzero()[2][i]
+        if aim == 'full':
+            if arr[x, y, z] < 30 and arr[x, y, z] != 1:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+        elif aim == 'c2':
+            if arr[x, y, z] == 3:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+    list_label_image.sort(key=lambda x: x[3])
+    return list_label_image
+
+
+def get_midslice_average(path_im, ind):
+    """
+    Retrieve the input images for the network. This images are generated by
+    averaging the 7 slices in the middle of the volume
+    Args:
+        path_im(string): path to image
+        ind(int): index of the slice around which we will average
+
+    Returns:
+        array: an array containing the average image oriented according to RAS+ convention.
+
+    """
+    image = nib.load(path_im)
+    image = nib.as_closest_canonical(image)
+    arr = np.array(image.dataobj)
+    numb_of_slice = 3
+    if ind + 3 > arr.shape[0]:
+        numb_of_slice = arr.shape[0] - ind
+    if ind - numb_of_slice < 0:
+        numb_of_slice = ind
+
+    return np.mean(arr[ind - numb_of_slice:ind + numb_of_slice, :, :], 0)
+
+
+def images_normalization(img_list, std=True):
+    if type(img_list) != list:
+        img_list = [img_list]
+    img_norm_list = []
+    for i in range(len(img_list)):
+        # print('Normalizing ' + str(i + 1) + '/' + str(len(img_list)))
+        img = img_list[i] - np.mean(img_list[i])  # zero-center
+        if std:
+            img_std = np.std(img)  # normalize
+            epsilon = 1e-100
+            img = img / (img_std + epsilon)  # epsilon is used in order to avoid by zero division
+        img_norm_list.append(img)
+    return img_norm_list
+
+
+def extract_mid_slice_and_convert_coordinates_to_heatmaps(bids_path, suffix, aim, ap_pad=128, is_pad=320):
+    """
+     This function takes as input a path to a dataset and generate two sets of images:
+   (i) mid-sagittal image of common size (1,ap_pad,is_pad) and
+   (ii) heatmap of disc labels associated with the mid-sagittal image.
+
+    Args:
+        bids_path (string): path to BIDS dataset form which images will be generated
+        suffix (string): suffix of image that will be processed (e.g., T2w)
+        aim(string): 'full' or 'c2'. If 'c2' retrieves only c2 label (value = 3) else create heatmap with all label.
+        ap_pad(int): desired output size of AP axis which will be
```

Comment:
try to avoid using MRI-specific orientation terms (here: AP), anticipating accommodation to histology